tests/lib/rules/space-between-siblings.js

```diff
@@ -0,0 +1,384 @@
+        },
+        {
+          message: 'Expected blank line after this block.',
+          line: 15,
+          column: 9
+        }
+      ]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <template>
+        <div>
+          <ul>
+            <li>
+            </li>
+            <li>
+            </li>
+            <br />
+            <li>
+            </li>
+          </ul>
+        </div>
+      </template>
+      `,
+      output: `
+      <template>
+        <div>
+          <ul>
+            <li>
+            </li>
+
+            <li>
+            </li>
+
+            <br />
+            <li>
+            </li>
+          </ul>
+        </div>
+      </template>
+      `,
+      errors: [
+        {
+          message: 'Expected blank line after this block.',
+          line: 7,
+          column: 13
+        },
+        {
+          message: 'Expected blank line after this block.',
+          line: 9,
+          column: 13
+        }
+      ],
+      options: [{ ignoreNewlinesAfter: ['br'] }]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <template>
+        <div>
+          <ul>
+            <li>
+            </li>
+            <li>
+            </li>
+            <br />
+            <li>
+            </li>
+          </ul>
+        </div>
+      </template>
+      `,
+      output: `
+      <template>
+        <div>
+          <ul>
+            <li>
+            </li>
+
+            <li>
+            </li>
+            <br />
+
+            <li>
+            </li>
+          </ul>
+        </div>
+      </template>
+      `,
+      errors: [
+        {
+          message: 'Expected blank line after this block.',
+          line: 7,
+          column: 13
+        },
+        {
+          message: 'Expected blank line after this block.',
+          line: 9,
+          column: 13
+        },
+        {
+          message: 'Expected blank line after this block.',
+          line: 10,
+          column: 13
+        }
+      ],
+      options: [{ ignoreNewlinesBefore: ['br'] }]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <template>
+        <div>
+          <ul>
+            <li>
+            </li>
+            <li>
+            </li>
+            <br />
+            <li>
+            </li>
+          </ul>
+        </div>
+      </template>
+      `,
+      output: `
+      <template>
+        <div>
+          <ul>
+            <li>
+            </li>
+
+            <li>
+            </li>
+            <br />
+            <li>
+            </li>
+          </ul>
+        </div>
+      </template>
+      `,
+      errors: [
+        {
+          message: 'Expected blank line after this block.',
+          line: 7,
+          column: 13
+        },
+        {
+          message: 'Expected blank line after this block.',
+          line: 9,
+          column: 13
+        }
+      ],
+      options: [{ ignoreNewlinesBefore: ['br'], ignoreNewlinesAfter: ['br'] }]
+    }
+  ]
```

Comment:
What happens for this code?

```html
<template>
  <p>Foo <b>bar</b> baz.</p>
</template>
```

I think it should be valid. Please add a test case for that.